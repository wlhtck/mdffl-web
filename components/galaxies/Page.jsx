import React from 'react';
import { map, get } from 'lodash/fp';
import ContentBlock from '../constellations/ContentBlock';
import HeroBlock from '../constellations/HeroBlock';

const ContentComponents = {
  heroBlock: HeroBlock,
  contentBlock: ContentBlock,
};

const Page = ({ data }) => (
  map((content) => {
    const Component = ContentComponents[get('type')(content)];

    return (
      <Component {...content} />
    );
  })(data)
);

export default Page;
