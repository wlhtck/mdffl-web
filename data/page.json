{"content":[{"sys":{"space":{"sys":{"type":"Link","linkType":"Space","id":"8wpi5cmhr3lg"}},"id":"3mpaJxFaJqAwoeSEse46QA","type":"Entry","createdAt":"2018-11-03T14:51:21.670Z","updatedAt":"2018-11-03T14:51:21.670Z","environment":{"sys":{"id":"master","type":"Link","linkType":"Environment"}},"revision":1,"contentType":{"sys":{"type":"Link","linkType":"ContentType","id":"contentBlock"}},"locale":"en-US"},"fields":{"header":"Hello world","body":"This is a test"}}],"slug":"index"}