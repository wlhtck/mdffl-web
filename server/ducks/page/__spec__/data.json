{
	"items": [{
		"fields": {
			"content": [{
				"sys": {
					"contentType": {
						"sys": {
							"type": "Link",
							"linkType": "ContentType",
							"id": "heroBlock"
						}
					},
					"locale": "en-US"
				},
				"fields": {
					"headline": "The bear is a hero!",
					"image": {
						"fields": {
							"title": "Bear sitting",
							"file": {
								"url": "//images.ctfassets.net/8wpi5cmhr3lg/PLDH0id61xTjbGlkiuEKU/5aa125d518233d0efc2f4a2ea1dabb0e/mark-basarab-723939-unsplash.jpg",
								"details": {
									"size": 3858763,
									"image": {
										"width": 4352,
										"height": 2904
									}
								},
								"fileName": "mark-basarab-723939-unsplash.jpg",
								"contentType": "image/jpeg"
							}
						}
					},
					"body": "The bear is a big hero. Everyone is super thankful!",
					"cta": {
						"fields": {
							"text": "Follow the bear!",
							"url": "https://www.facebook.com/groups/bear.inc/",
							"external": true
						}
					},
					"slug": "hero-bear"
				}
			}, {
				"sys": {
					"contentType": {
						"sys": {
							"type": "Link",
							"linkType": "ContentType",
							"id": "contentBlock"
						}
					},
					"locale": "en-US"
				},
				"fields": {
					"headline": "Check out the bear!",
					"body": "The bear is super cool and everyone loves him! Click below to find out more!",
					"imageAlign": "right",
					"image": {
						"fields": {
							"title": "Bear",
							"file": {
								"url": "//images.ctfassets.net/8wpi5cmhr3lg/6mW6Tsx5scOYMwocwSskKq/f1bcb0a3f979f7de3e1ba070d0d14f4f/photo-1541560584704-f2767960951a.jpeg",
								"details": {
									"size": 1769907,
									"image": {
										"width": 1856,
										"height": 2830
									}
								},
								"fileName": "photo-1541560584704-f2767960951a.jpeg",
								"contentType": "image/jpeg"
							}
						}
					},
					"cta": {
						"fields": {
							"text": "Google",
							"url": "https://www.google.com/search?q=Bear",
							"external": true
						}
					},
					"slug": "bear"
				}
			}],
			"slug": "index",
			"name": "Home"
		}
	}, {
		"fields": {
			"content": [{
				"sys": {
					"contentType": {
						"sys": {
							"type": "Link",
							"linkType": "ContentType",
							"id": "contentBlock"
						}
					},
					"locale": "en-US"
				},
				"fields": {
					"headline": "Check out the bear!",
					"body": "The bear is super cool and everyone loves him! Click below to find out more!",
					"imageAlign": "right",
					"image": {
						"fields": {
							"title": "Bear",
							"file": {
								"url": "//images.ctfassets.net/8wpi5cmhr3lg/6mW6Tsx5scOYMwocwSskKq/f1bcb0a3f979f7de3e1ba070d0d14f4f/photo-1541560584704-f2767960951a.jpeg",
								"details": {
									"size": 1769907,
									"image": {
										"width": 1856,
										"height": 2830
									}
								},
								"fileName": "photo-1541560584704-f2767960951a.jpeg",
								"contentType": "image/jpeg"
							}
						}
					},
					"cta": {
						"fields": {
							"text": "Google",
							"url": "https://www.google.com/search?q=Bear",
							"external": true
						}
					},
					"slug": "bear"
				}
			}, {
				"sys": {
					"contentType": {
						"sys": {
							"type": "Link",
							"linkType": "ContentType",
							"id": "contentBlock"
						}
					},
					"locale": "en-US"
				},
				"fields": {
					"headline": "Check out the bear!",
					"body": "The bear is super cool and everyone loves him! Click below to find out more!",
					"imageAlign": "left",
					"image": {
						"fields": {
							"title": "Bear",
							"file": {
								"url": "//images.ctfassets.net/8wpi5cmhr3lg/6mW6Tsx5scOYMwocwSskKq/f1bcb0a3f979f7de3e1ba070d0d14f4f/photo-1541560584704-f2767960951a.jpeg",
								"details": {
									"size": 1769907,
									"image": {
										"width": 1856,
										"height": 2830
									}
								},
								"fileName": "photo-1541560584704-f2767960951a.jpeg",
								"contentType": "image/jpeg"
							}
						}
					},
					"cta": {
						"fields": {
							"text": "Google",
							"url": "https://www.google.com/search?q=Bear",
							"external": true
						}
					},
					"slug": "bear-2"
				}
			}],
			"slug": "about",
			"name": "About"
		}
	}]
}
