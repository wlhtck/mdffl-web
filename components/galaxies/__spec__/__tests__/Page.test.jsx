import React from 'react';
import { shallow } from 'enzyme';
import Page from '../../Page';
import data from '../page.data';

describe('components/galaxies/Page', () => {
  it('renders <HeroBlock />s and <ContentBlock />s', () => {
    expect(shallow(<Page data={data.page} />)).toMatchSnapshot();
  });
});
