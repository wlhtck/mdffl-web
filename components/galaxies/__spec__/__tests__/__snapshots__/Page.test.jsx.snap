// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/galaxies/Page renders <HeroBlock />s and <ContentBlock />s 1`] = `
Array [
  <HeroBlock__StyledHeroBlock
    body="The bear is a big hero. Everyone is super thankful!"
    cta={
      Object {
        "external": true,
        "text": "Google",
        "url": "https://google.com",
      }
    }
    headline="The bear is a hero!"
    imgSrc="https://images.unsplash.com/photo-1530595467537-0b5996c41f2d?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1440&q=80"
    type="heroBlock"
  />,
  <ContentBlock__StyledContentBlock
    background={false}
    body="The bear is super cool and everyone loves him! Click below to find out more!"
    cta={
      Object {
        "external": true,
        "text": "Google",
        "url": "https://google.com",
      }
    }
    headline="Check out the bear!"
    textAlign="center"
    type="contentBlock"
  />,
  <ContentBlock__StyledContentBlock
    background={false}
    body="The bear is super cool and everyone loves him! Click below to find out more!"
    cta={
      Object {
        "external": true,
        "text": "Google",
        "url": "https://google.com",
      }
    }
    headline="Check out the bear!"
    image={
      Object {
        "alt": "Bear",
        "src": "https://images.unsplash.com/photo-1541560584704-f2767960951a?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1856&q=80",
      }
    }
    imageAlign="right"
    textAlign="left"
    type="contentBlock"
  />,
]
`;
